cryptol-saw-core/src/Verifier/SAW/CryptolEnv.hs

```diff
@@ -544,18 +548,12 @@ parseDecls sc env input = do
     -- Convert from 'Decl' to 'TopDecl' so that types will be generalized
     let topdecls = [ P.Decl (P.TopLevel P.Public Nothing d) | d <- npdecls ]
 
-    -- Label each TopDecl with the "interactive" module for unique name generation
-    let (mdecls :: [MN.InModule (P.TopDecl P.PName)]) = map (MN.InModule interactiveName) topdecls
-    nameEnv1 <- MN.liftSupply (MN.namingEnv' mdecls)
-
     -- Resolve names
-    let nameEnv = nameEnv1 `MR.shadowing` getNamingEnv env
-    (rdecls :: [P.TopDecl T.Name]) <- MM.interactive (MB.rename interactiveName nameEnv (traverse MR.rename topdecls))
+    (_nenv, rdecls) <- MM.interactive (MB.rename interactiveName (getNamingEnv env) (MR.renameTopDecls interactiveName topdecls))
```

Comment:
@robdockins I am  not really sure, but perhaps there is a problem here.   Note that in the old code the naming environment in which we do the resolving already has the names added to it, while in the new one it does not...

Otoh, I don't understand why the old code was doing that in the first place, and what you've replaced it with makes sense to me, so perhaps the problem is not here.

To be more help, I should probably just check out the code and run it, ping me on the chat if that'd be useful.

Reply:
I wondered about this code too, but I think it's unrelated, actually.  I think this code is for dealing with declarations appearing inside the `{{ ... }}` delimiters in SAWScript.  The code for importing modules is further above, and is barely changed.